### DISCUSSION TITLE
function not accessible in #include'd file

### DISCUSSION THREAD

**Discussion by kolja** [typst/typst#1224] [Q&A] [answered]
I have the following document:
```
#let highlight(word) = [
    #set text(yellow)
    #word
]

#show emph: it => [
    #set text(green)
    #it.body
]

#include "chapter1.typ"
#include "chapter2.typ"
#include "chapter3.typ"
```
I would like to highlight/emphasize certain words in my chapters.
I am able to do this with the ``emph`` Rule (by surrounding words with underscores), 
However, when I try to use ``#highlight[word]`` in an included chapter, compilation fails (error: unknown variable).

Apparently variables are not exposed to the included files (or am I missing something)?
Is there a way to make variables available for all included chapters?

**Comment by laurmaedje** [comment, marked as answer]
You should extract `highlight` into a third file from which you import it in all files where you need it. See also #754 and #595.

**Comment by ondohotola** [comment]
I am also struggling with that

I am using [tablex](https://github.com/PgBiel/typst-tablex) from the wiki, which is something like 2500 lines. 

I need to rewrite a reasonably large handbook with something like 25 chapters (`3include "chapter.typ"`) and don't really want to import it each time into each file (25 times) but would obviously prefer to do this only once.

Will something like this be implemented?

**Reply by ondohotola** [reply]
But seeing the discussion elsewhere it’ll probably not happen. On the other hand the loading is very fast and bound to be cached, so no big drama.

**Reply by kolja** [reply]
I was also expecting `include` to work the same as `\input{filename}` in TeX: 
That the current context would become available to the included file. 
People splitting up their documents into separate files/chapters seems to be the most common use case for this feature.

I fiddled around with `eval` and `read` a little...
```typst
#let insert(file, context) = {
  eval(read(context) + "\n[" + read(file) + "]")
}

#insert("chapter.typ", "context.typ")
```
but this is not the way.
I'll stick with @laurmaedje 's suggestion and import stuff where it's needed.
